### DISCUSSION TITLE
playing audio and video for video-keyboard-response

### DISCUSSION THREAD

**Discussion by michelle-min** [jspsych/jsPsych#952] [Q&A] [answered]
Post #841 mentioned that loose synchronization of audio and video is possible with jsPsych. What plugins/customization could I use if I wanted to play separate audio and video files simultaneously at the start of a trial? I don't need to record any specific responses to the stimuli, just show them together for the full length of the video. I'm currently using video-keyboard-response to show the videos. Thanks!

**Comment by jodeleeuw** [comment, marked as answer]
Hi @michelleymin,

You'll need to modify the `video-keyboard-response` plugin to do this. 

You can borrow bits of code from the `audio-keyboard-response` plugin to get the audio working.

This code sets up the stimulus, allowing for either HTML5 or WebAudioAPI playback:
https://github.com/jspsych/jsPsych/blob/83980085ef604c815f0d97ab55c816219e969b84/plugins/jspsych-audio-keyboard-response.js#L63-L72

This code starts the audio playback:
https://github.com/jspsych/jsPsych/blob/83980085ef604c815f0d97ab55c816219e969b84/plugins/jspsych-audio-keyboard-response.js#L146-L152

This code stops the audio if the trial ends for another reason:
https://github.com/jspsych/jsPsych/blob/master/plugins/jspsych-audio-keyboard-response.js#L103-L111

You'll also need to setup the parameters in the plugin.info, and call the preload method:

https://github.com/jspsych/jsPsych/blob/83980085ef604c815f0d97ab55c816219e969b84/plugins/jspsych-audio-keyboard-response.js#L17-L26

https://github.com/jspsych/jsPsych/blob/83980085ef604c815f0d97ab55c816219e969b84/plugins/jspsych-audio-keyboard-response.js#L15

That should be the main bits of code you'll need to add. I'm sure there will be some small complications with integration, but it shouldn't be too bad. Feel free to follow up here with any issues you encounter.

**Reply by michelle-min** [reply]
That worked perfectly, thank you!
